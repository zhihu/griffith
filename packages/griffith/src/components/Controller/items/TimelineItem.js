import React from 'react'
import {css} from 'aphrodite/no-important'
import styles from '../styles'
import Timeline from '../../Timeline'

const TimelineItem = props => {
  return (
    <div className={css(styles.timeline)}>
      <Timeline {...props} />
    </div>
  )
}

export default TimelineItem
