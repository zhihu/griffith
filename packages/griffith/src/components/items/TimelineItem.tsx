import React from 'react'
import {css} from 'aphrodite/no-important'
import styles from '../Controller.styles'
import Timeline, {TimelineProps} from '../Timeline'

const TimelineItem = (props: TimelineProps) => {
  return (
    <div className={css(styles.timeline)}>
      <Timeline {...props} />
    </div>
  )
}

export default TimelineItem
