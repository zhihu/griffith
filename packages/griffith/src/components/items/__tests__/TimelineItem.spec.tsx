import React from 'react'
import {shallow} from 'enzyme'
import TimelineItem from '../TimelineItem'

describe('TimelineItem', () => {
  it('get TimelineItem component', () => {
    const wrapper = shallow(
      <TimelineItem
        value={7}
        total={187}
        buffered={12}
        onDragStart={jest.fn()}
        onDragEnd={jest.fn()}
        onChange={jest.fn()}
        onSeek={jest.fn()}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
