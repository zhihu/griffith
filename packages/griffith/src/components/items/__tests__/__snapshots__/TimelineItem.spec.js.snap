// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimelineItem get TimelineItem component 1`] = `
<div
  className="timeline_bc4egv"
>
  <Timeline
    buffered={12}
    onChange={[MockFunction]}
    onDragEnd={[MockFunction]}
    onDragStart={[MockFunction]}
    onSeek={[MockFunction]}
    total={187}
    value={7}
  />
</div>
`;
